src/hooks/useBreakpoint.ts

```diff
@@ -25,19 +25,27 @@ const useBreakpoint = (
     const [currentBreakpoints, setCurrentBreakpoints] = useState<string[]>([]);
     const nodeRef = useRef<HTMLDivElement>(null);
 
-    const performCheckSize = () => {
-        window.requestAnimationFrame(() => {
+    const performCheckSize = async (abortSignal: AbortSignal) => {
+        enqueueAsyncOperation(() => {
             if (!nodeRef.current) {
+                performCheckSize(abortSignal);
```

Comment:
Why is this recursive call neseccary?

Reply:
Good point. The useEffect with dependency on the nodeRef will restart the size checking